test/Identity.FunctionalTests/Pages/Index.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Net.Http;
+using System.Threading.Tasks;
+using AngleSharp.Dom.Html;
+using Microsoft.AspNetCore.Identity.FunctionalTests.Account;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Identity.FunctionalTests
+{
+    public class Index : HtmlPage
+    {
+        private readonly bool _authenticated;
+        private readonly IHtmlAnchorElement _registerLink;
+        private readonly IHtmlAnchorElement _loginLink;
+        //private readonly IHtmlAnchorElement _logout;
+        private readonly IHtmlAnchorElement _manageLink;
+        public static readonly string Path = "/";
+
+        public Index(HttpClient client, IHtmlDocument index, HtmlPageContext context, bool authenticated)
+            : base(client, index, context)
+        {
+            _authenticated = authenticated;
+            if (!_authenticated)
+            {
+                _registerLink = HtmlAssert.HasLink("#register", Document);
```

Comment:
This should ensure the opposite is true when authenticated i,e, `#register` does not appear for authenticated users

Reply:
That increases the complexity of the tests a lot without providing too much value. In the case where it's worth it, we can have it, but I think it's ok to focus on the elements that should be present on the page rather than on the ones that are not present.